interfaces/builtin/mount_control.go

```diff
@@ -0,0 +1,469 @@
+		}
+
+		if err := fn(mountInfo); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func validateWhatAttr(what string) error {
+	if !whatRegexp.MatchString(what) {
+		return fmt.Errorf(`mount-control "what" attribute is invalid: must start with / and not contain special characters`)
+	}
+
+	if !cleanSubPath(what) {
+		return fmt.Errorf(`mount-control "what" pattern is not clean: %q`, what)
+	}
+
+	if _, err := utils.NewPathPattern(what); err != nil {
+		return fmt.Errorf(`mount-control "what" setting cannot be used: %v`, err)
+	}
+
+	return nil
+}
+
+func validateWhereAttr(where string) error {
+	if !whereRegexp.MatchString(where) {
+		return fmt.Errorf(`mount-control "where" attribute is invalid: must start with $SNAP_COMMON, $SNAP_DATA or /media and not contain special characters`)
+	}
+
+	if !cleanSubPath(where) {
+		return fmt.Errorf(`mount-control "where" pattern is not clean: %q`, where)
+	}
+
+	if _, err := utils.NewPathPattern(where); err != nil {
+		return fmt.Errorf(`mount-control "where" setting cannot be used: %v`, err)
+	}
+
+	return nil
+}
+
+func validateMountTypes(types []string) error {
+	for _, t := range types {
+		if !typeRegexp.MatchString(t) {
+			return fmt.Errorf(`mount-control filesystem type invalid: %q`, t)
+		}
+		if !strutil.ListContains(allowedFSTypes, t) {
+			return fmt.Errorf(`mount-control forbidden filesystem type: %q`, t)
+		}
+	}
+	return nil
+}
+
+func validateMountOptions(options []string) error {
+	if len(options) == 0 {
+		return errors.New(`mount-control "options" cannot be empty`)
+	}
+	for _, o := range options {
+		if !strutil.ListContains(allowedMountOptions, o) {
+			return fmt.Errorf(`mount-control option unrecognized or forbidden: %q`, o)
+		}
+	}
+	return nil
+}
+
+// Find the first option which is incompatible with a FS type declaration
+func optionIncompatibleWithFsType(options []string) string {
+	for _, o := range options {
+		if strutil.ListContains(optionsWithoutFsType, o) {
+			return o
+		}
+	}
+	return ""
+}
+
+func validateMountInfo(mountInfo *MountInfo) error {
+	if err := validateWhatAttr(mountInfo.what); err != nil {
+		return err
+	}
+
+	if err := validateWhereAttr(mountInfo.where); err != nil {
+		return err
+	}
+
+	if err := validateMountTypes(mountInfo.types); err != nil {
+		return err
+	}
+
+	if err := validateMountOptions(mountInfo.options); err != nil {
+		return err
+	}
+
+	// Check if any options where given which are incompatible with specifying
+	// a FS type
+	fsExclusiveOption := optionIncompatibleWithFsType(mountInfo.options)
+	if fsExclusiveOption != "" && len(mountInfo.types) > 0 {
+		return fmt.Errorf(`mount-control option %q is incompatible with a FS type declaration`, fsExclusiveOption)
+	}
+
+	return nil
+}
+
+func (iface *mountControlInterface) BeforeConnectPlug(plug *interfaces.ConnectedPlug) error {
+	// The systemd.ListMountUnits() method works by issuing the command
+	// "systemctl show *.mount", but globbing was only added in systemd v209.
```

Comment:
True, this bit is not strictly necessary as of now, but suppose that we release a new snapd with this PR and not with the next one, it would be sad if someone started using it on Trusty and then it stopped working when the final bit land.

(it's a rather unlikely corener-case, since the store must approve the usage of this interface, but still :-p)